jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -228,14 +228,15 @@ private void analyzeFile(String pkgName, String inPath, boolean includeNonPublic
     } else if (!new File(inPath).exists()) {
       return;
     }
-    AnalysisScope scope = AnalysisScopeReader.makePrimordialScope(null);
+    AnalysisScope scope = AnalysisScopeReader.instance.makePrimordialScope(null);
```

Comment:
Staring at the global mutable static field access with deep suspicion 👀 ...  but all `AnalysisScopeReader.instance` itself seems to contain is a link to its own class loader and a constant string, unless subclassed, so I assume this will be safe.

Reply:
Yes this is gross. I will fix on the WALA side. I think the `AnalysisScopeReader` constructor can just be public. But I'll have to look more closely. In the meantime yes I think the current code is thread safe since the type doesn't have any mutable state

Reply:
Seems like subclasses can [set themselves as the instance](https://github.com/wala/WALA/blob/master/com.ibm.wala.core/src/main/java/com/ibm/wala/core/util/config/AnalysisScopeReader.java#L42), and that would be a reason for the pattern: but it's risky if subclasses aren't very careful about what they contain and when they call this `setScopeReader(...)` method.

Reply:
Yeah, I think in upstream WALA if I make the `AnalysisScopeReader` constructor public, we can just change `JarInfer` to call it directly and not rely on the static state.  But will go with this for now